app/app.ts

```diff
@@ -437,7 +437,7 @@ function setupIpc(localSettings: LocalSettings, scrSettings: ScrSettings) {
         try {
           shieldBatteryData = parseShieldbatteryReplayData(buffer)
         } catch (err) {
-          reject(err)
+          logger.error(`Error parsing the replay's ShieldBattery data section: ${err}`)
```

Comment:
You should really include the `stack` from errors, we use this pattern most places:
`${err.stack ?? err}`